api/src/security/auth-utils.test.ts

```diff
@@ -270,7 +278,161 @@ describe('authorize', function () {
       await auth_utils.authorize({ keycloak_token: 'some token' }, ['abc']);
       expect.fail();
     } catch (actualError) {
-      expect(actualError.message).to.contain('Access Denied');
+      expect(actualError).instanceOf(HTTP403);
+    }
+  });
+
+  it('throws HTTP403 when stubbed getSystemUser throws error', async function () {
+    sinon.stub(auth_utils, 'getSystemUser').callsFake(async function (keycloak_token: object) {
+      if (!keycloak_token || keycloak_token) {
+        throw new Error();
+      }
+    });
+
+    try {
+      await auth_utils.authorize({ keycloak_token: 'any token' }, ['abc']);
+      expect.fail();
+    } catch (actualError) {
+      expect(actualError).instanceOf(HTTP403);
+    }
+  });
+
+  it('throws HTTP403 when stubbed getSystemUser returns user with no roles', async function () {
```

Comment:
nit: technically this test is checking that an error is thrown when `userHasValidSystemRoles` returns falsie.